lib/cldr/export/data/numbers.rb

```diff
@@ -7,108 +7,181 @@ class Numbers < Base
-            format_key = format_key ? format_key.value : :default
+          aliased = select_single(number_system_node, "alias")
+          if aliased
+            return xpath_to_format_alias(aliased["path"], type)
+          end
 
-            if !format_nodes.empty?
-              format_nodes.each do |format_node|
-                format_result[format_key] ||= select(format_node, "pattern").each_with_object({}) do |pattern_node, pattern_result|
-                  pattern_key_node = pattern_node.attribute("type")
+          result = select("numbers/#{type}Formats[@numberSystem=\"#{number_system}\"]/#{type}FormatLength").each_with_object({}) do |format_length_node, format_result|
+            format_length_key = format_length_node["type"]&.to_sym || default_format_length_type
 
-                  pattern_count_node = pattern_node.attribute("count")
+            aliased = select_single(format_length_node, "alias")
+            if aliased
+              format_result[format_length_key] = xpath_to_redirect(aliased["path"], number_system, type)
+              next
+            end
 
-                  pattern_key = pattern_key_node ? pattern_key_node.value : :default
+            format_result[format_length_key] = if format_length_key == default_format_length_type
+              parse_default_currency_format_length_node(number_system, format_length_node, type)
+            else
+              parse_currency_format_length_node(format_length_node, type)
+            end
+          end
 
-                  if pattern_count_node
-                    pattern_count = pattern_count_node.value
+          result
+        end
 
-                    if pattern_result[pattern_key].nil?
-                      pattern_result[pattern_key] ||= {}
-                    elsif !pattern_result[pattern_key].is_a?(Hash)
-                      raise "can't parse patterns with and without 'count' attribute in the same section"
-                    end
+        def parse_default_currency_format_length_node(number_system, format_length_node, type)
+          result = {}
+          select(format_length_node, "#{type}Format").each do |format_node|
+            format_key = format_node["type"]&.to_sym || default_format_type
 
-                    pattern_result[pattern_key][pattern_count] = pattern_node.content
-                  else
-                    pattern_result[pattern_key] = pattern_node.content
-                  end
-                end
-              end
+            if (aliased = select_single(format_node, "alias"))
+              format_length_key = format_length_node["type"]&.to_sym || default_format_length_type
+              result[format_key] = xpath_to_default_currency_format_length_node_alias(aliased["path"], number_system, format_length_key)
             else
-              aliased = select_single(format_length_node, "alias")
+              pattern_nodes = select_single(format_node, "pattern")
+              result[format_key] = pattern_nodes.content
```

Comment:
I notice that there can be multiple `<pattern>` nodes in this scenario, some with `alt` attributes. I'm assuming that it's intentional here to only select the first one?

```suggestion
              pattern_node = select_single(format_node, "pattern")
              result[format_key] = pattern_node.content
```

Reply:
I saw it in [`common/main/agq.xml`](https://github.com/unicode-org/cldr/blob/main/common/main/agq.xml#L611), though these also have the attributes of `draft="provisional"`.

I also see some examples in [`common/main/bn_IN.xml`](https://github.com/unicode-org/cldr/blob/main/common/main/bn_IN.xml#L657) (also, some examples of `↑↑↑` as values, which I'm assuming are inheritance markers).

Reply:
> I also searched CLDR for `<pattern.*alt`
Oh I see. It seems my local clone wasn't up to date. 😅
Thanks for pointing these out.

Reply:
Changed xpath query to ignore those `alt` nodes for now.